tests/integration/test_charm.py

```diff
@@ -102,6 +102,7 @@ async def driver(request, ops_test):
         driver.get_screenshot_as_file(f"/tmp/selenium-{request.node.name}.png")
 
 
+@pytest.mark.asyncio
```

Comment:
Why is this is needed now and not before.

Reply:
Seems like something changed about our environment.  There's a few places where these tests now fail.  Maybe it is a pytest-operator change, or something about the runner?  I'm also confused

Reply:
This might be because of the [pytest-asyncio v0.19.0 release](https://github.com/pytest-dev/pytest-asyncio/releases/tag/v0.19.0).  They changed the default value of `asyncio_mode` from `auto` to `strict`:
* `auto` mode assumed that any async test should be handled by pytest-asyncio, even if it doesn't have the `@pytest.mark.asyncio` marker.  
* `strict` mode requires that you explicitly use the `@pytest.mark.asyncio` marker, otherwise the test wont be async

We can handle this by either adding `@pytest.mark.asyncio` to all our tests, or by switching back to `auto` mode, either through `pytest ... --asyncio-mode=auto` or adding this to a `pytest.ini` file:
```
# pytest.ini
[pytest]
asyncio_mode = auto
```

Reply:
I think it would make sense to add the `asyncio_mode=auto` to our config for all our repos.  Thoughts?  `strict` is mentioned as something that's necessary when you have multiple async runners, but not when you're using just one

Reply:
Okay, makes sense. In this PR we have both `@pytest.mark.asyncio` ONLY for this test case and `--asyncio-mode=auto` in `tox.ini`. Should we just keep the change in `tox.ini` and move on? I guess that would be easier to change in all our charms (the ones missing it) given what you just mentioned.

Reply:
I'm more confused now.  [pytest-operator says it automatically applies the async mark already](https://github.com/charmed-kubernetes/pytest-operator/blob/main/docs/reference.md#pytestmarkasyncio)

Reply:
maybe because ops is pinned to 1.2

Reply:
I think I understand it more now. [this issue](https://github.com/canonical/bundle-kubeflow/issues/470) is a knowledge dump of what I think is happening.

TL/DR:
* new `pytest-asyncio` version `v0.19.0` has changed from `asyncio-mode=auto` to `asyncio-mode=strict`
  * `asyncio-mode=auto` makes `pytest-asyncio` assume any test *and fixture* is async, and handle it accordingly
  * `asyncio-mode=strict` means the user must do extra work to handle async tests and fixtures
* our driver fixture uses `pytest.fixture()` instead of `pytest_asyncio.fixture()`, resulting in the generator error we observed in the CI
* the `test_links` test function does not have a `@pytest.mark.asyncio` mark in our code, which normally would also cause an error in `asyncio-mode=strict`.  But, `pytest-operator` has the extra behaviour of automatically marking every test as async for you as a convenience, so we're covered anyway

Options to fix this:
1. go back to `auto`, either by adding `--asyncio-mode=auto` in the `pytest` call as @agathanatasha had earlier, or by adding it to the `pyproject.toml`, `tox.ini`, or similar
2. add the required async fixture decorators so that `strict` just works.  

I think it would be easier to do option 2, but I don't care much either way.